src/renderer/components/Text.tsx

```diff
@@ -0,0 +1,127 @@
+import { CSSProperties, PropsWithChildren } from 'react'
+import Typography from '@mui/material/Typography'
+
+type BaseProps = PropsWithChildren<{
+  align?: Extract<
+    CSSProperties['textAlign'],
+    'inherit' | 'left' | 'center' | 'right' | 'justify'
+  >
+  bold?: boolean
+  className?: string
+  id?: string
+  italic?: boolean
+  style?: CSSProperties
+  underlined?: boolean
+}>
+
+export function TitleText({
+  align,
+  bold,
+  children,
+  className,
+  italic,
+  style,
+  underlined,
+}: BaseProps) {
+  return (
+    <Typography
+      variant="h1"
+      align={align}
+      style={{
+        ...style,
+        fontWeight: bold ? 'bold' : undefined,
+        fontStyle: italic ? 'italic' : undefined,
+        textDecoration: underlined ? 'underline' : undefined,
```

Comment:
This means that you if you pass the style prop `{ fontVariant: 'bold' }`, this will be overridden with `undefined` for the prop passed to `Typography`. Is this desired behaviour?

Reply:
just to clarify, do you mean `fontWeight` in the example you provided? ([font variant](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant) is a separate but unrelated css property) I'm assuming that's what you meant, and thus the answer is no, we should definitely attempt to fall back to the `style` prop if provided e.g.

```tsx
fontWeight: bold ? 'bold' : style?.fontWeight,`
```

an additional thing that should be clarified is the precedence of props like `bold` vs `style`. not sure how these libraries usually behave, but what should happen when you have something like this?

```tsx
<Text bold style={{ fontWeight: 'normal' }} />
```

Reply:
yeah fontWeight is what I meant. Precedence is tricky! Also, does MUI actually support the style prop? I feel like we should keep the wrapper on MUI fairly light right now - my main question would be "how does MUI handle bold"?

Reply:
> does MUI actually support the style prop?

it supports a prop called [`sx`](https://mui.com/system/getting-started/the-sx-prop/) which is tied to style overrides via emotion's styled components. we opted to map our `style` prop to that in the button component (see https://github.com/digidem/comapeo-desktop/pull/27#discussion_r1664589542) with the intention of rarely using it

Reply:
ah it also supports a `style` prop, which gets passed all the way down to the html element without any of the fanciness of that `sx` prop

Reply:
> my main question would be "how does MUI handle bold"?

looks like both `style` and `sx` override any of the more direct props in MUI, so i guess maybe just stick with that behavior? (which seems to make enough sense in general)

Reply:
When you say "stick with that", you mean stick with using style to override? @achou11 ?

Reply:
yep! hoping this [comment](https://github.com/digidem/comapeo-desktop/pull/28#discussion_r1674436563) clarifies a bit but yeah I think a reasonable rule of thumb for any components that expose a `style` prop would be that it overrides any props that may be affected by that `style` prop (in our case, `bold`, `italic`, and `underlined`)